librz/core/cmd/cmd_resize.c

```diff
@@ -4,21 +4,55 @@
 #include <string.h>
 #include <rz_core.h>
 
-static int rebase_helper(RzCore *core, ut64 oldbase, ut64 newbase) {
-	rz_debug_bp_rebase(core->dbg, oldbase, newbase);
-	rz_bin_set_baddr(core->bin, newbase);
-	rz_flag_move(core->flags, oldbase, newbase);
-	rz_core_bin_apply_all_info(core, rz_bin_cur(core->bin));
-	// TODO: rz_analysis_move :??
-	// TODO: differentiate analysis by map ranges (associated with files or memory maps)
-	return 0;
-}
+/**
+ * \brief rb - rebase all flags, binary, information, breakpoints, and analysis.
+ * \param core Rizin core.
+ * \param argc Count of command arguments.
+ * \param argv Vector containing command arguments.
+ * \return RZ_CMD_STATUS_OK or RZ_CMD_STATUS_ERROR.
+ */
+RZ_IPI RzCmdStatus rz_rebase_handler(RZ_BORROW RzCore *core, int argc, RZ_BORROW const char **argv) {
+
+	// get current file and current object
+	RzBinFile *bf = rz_bin_cur(core->bin);
+	if (!(bf && rz_file_exists(bf->file))) {
```

Comment:
```suggestion
	if (!bf) {
```

I think you don't need to check if the file exists. Was there any particular reason why you did that? AFAIK bf->file could also refer to a non physical file.